src/examples/earth_trojan.py

```diff
@@ -0,0 +1,96 @@
+"""
+Earth Trojan 2010 TK7
+=====================
+
+Following some of the analysis from the following papers:
+
+- `"2002 AA29: Earth's recurrent quasi-satellite?" - Paweł Wajer - Icarus - 2009 <https://doi.org/10.1016/j.icarus.2008.10.018>`_  
+- `"Dynamical evolution of Earth’s quasi-satellites: 2004 GU9 and 2006 FV35" - Paweł Wajer - Icarus - 2010 <https://doi.org/10.1016/j.icarus.2010.05.012>`_
+
+We can perform some dynamical analysis to see if the asteroid 2010 TK7 appears
+to be an Earth trojan.
+
+The end result of this is a plot which shows the current energy level of the
+orbit, if the current position of the object is in a gravitational "low", and
+the energy line (in red) is below the height of the walls of that low point, then
+the object is at least temporarily captured.
+
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import kete
+
+
+# Grab 2010 TK7 and propagate it to 2010
+state_a = kete.HorizonsProperties.fetch("2010 TK7", update_name=False).state
+state_a = kete.propagate_n_body([state_a], kete.Time.from_ymd(2010, 1, 1))[0]
+
+state_b = kete.spice.get_state("Earth", state_a.jd)
+# Earth mass in Solar masses
+mass_planet = 3.0435727639549377e-06
+
+# n_sigma is the number of step on the x axis in the final plot
+n_sigma = 500
+
+# m_steps is the number of steps in the numerical integration
+m_steps = 100
+
+
+jd = state_a.jd
+
+elem_a = state_a.elements
+elem_b = state_b.elements
+
+period_a = elem_a.orbital_period
+period_b = elem_b.orbital_period
+
+sigma_steps = np.linspace(0, 1, n_sigma + 1)[:-1]
+
+energy_vals = []
+for shift in sigma_steps:
+
+    total = 0
+    for frac in np.linspace(0, 1, m_steps + 1)[:-1]:
+        state_a = kete.propagate_two_body([state_a], jd + (frac + shift) * period_a)[0]
```

Comment:
this is not quite right.  I don't believe you want to do propagation here.  I think rather you want to sweep mean anomaly for the instantaneous orbits, then evaluate the state at each of those MA clockings.  Time would be a third axis to consider; e.g. make one plot for time=jd, then integrate the jd of the system a step forward and make another plot.

Reply:
i.e., I'm not sure that this and the following line maintain the same delta-longitude for all steps in m_steps when period_a and period_b aren't the same, and I think you need to maintain that spacing to get the energy correct